Panic when BYDAY has a '+' symbol

Panic when BYDAY has the '+' symbol

Tested with the following
"RRULE:FREQ=MONTHLY;INTERVAL=1;BYDAY=+1WE"

panic: Unknown Day
github.com/graham/rrule@v0.0.0-20200104195902-7382dbb0d849/lib.go:96

Expected: Not panic, and '+' symbol is ignored (remove symbol while still maintaining offset_dir in lib.go:380)

--- a/lib.go
+++ b/lib.go
@@ -380,6 +380,8 @@ func (rr *RecurringRule) handle_recur_rule_part(key string, value string) error 
 			if item[0] == '-' {
 				offset_dir = -1
 				item = item[1:]
+			} else if item[0] == '+' {
+				item = item[1:]
 			}
 
 			if unicode.IsDigit(rune(item[0])) && unicode.IsDigit(rune(item[1])) {

--- a/parse_test.go
+++ b/parse_test.go
@@ -186,3 +186,26 @@ func Test_Parse_set_until_from_forever(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func Test_Parse_PositiveIndicatorIgnored(t *testing.T) {
+	var withPos = "DTSTART;TZID=America/New_York:19970902T090000\n" +
+		"RRULE:FREQ=MONTHLY;BYDAY=+1MO,+1TU"
+	wpRule, err := AssertToStringMatchesInput(withPos)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+	}
+
+	var noPos = "DTSTART;TZID=America/New_York:19970902T090000\n" +
+		"RRULE:FREQ=MONTHLY;BYDAY=1MO,1TU"
+	npRule, err := AssertToStringMatchesInput(noPos)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+	}
+
+	if wpRule.String() != npRule.String() {
+		t.Log("positive('+') indicator not ignored")
+		t.Fail()
+	}
+}

